src/txmempool.cpp

```diff
@@ -469,11 +469,43 @@ void CTxMemPool::removeUnchecked(txiter it)
     cachedInnerUsage -= it->DynamicMemoryUsage();
     cachedInnerUsage -= memusage::DynamicUsage(mapLinks[it].parents) + memusage::DynamicUsage(mapLinks[it].children);
     mapLinks.erase(it);
+    if (!minedInBlock) {
+        if (trackRemovalsCount) {
+            removedTxs.emplace_back(it->GetSharedTx());
+        }
+        else {
+            // If there are ever intentionally removals that are not
+            // meant to be tracked (so they can be notified on), then
+            // this log message can be removed.
+            LogPrint("mempool", "MemPoolRemovalTracker not tracking removed transaction %s\n",hash.ToString());
+        }
+    }
     mapTx.erase(it);
     nTransactionsUpdated++;
     minerPolicyEstimator->removeTx(hash);
 }
 
+void CTxMemPool::startTrackingRemovals()
+{
+    LOCK(cs);
+    trackRemovalsCount++;
+}
+
+std::vector<CTransactionRef>  CTxMemPool::stopTrackingRemovals() {
+    LOCK(cs);
+    assert(trackRemovalsCount);
+    trackRemovalsCount--;
+    if (trackRemovalsCount == 0) {
+        std::vector<CTransactionRef> temp(std::move(removedTxs));
+        removedTxs.clear();
+        return std::move(temp);
```

Comment:
It's better to have a single return statement that returns a local variable over using std::move:

```c++
    ...
    std::vector<CTransactionRef> temp;
    if (trackRemovalsCount == 0) {
        temp.emplace_back(std::move(removedTxs));
        removedTxs.clear();
    }
    return temp;
}
```

This allows the compiler to use NVRO (named return value optimization) - the `temp` variable essentially becomes an alias for the returned object.